pilot/pkg/networking/util/util.go

```diff
@@ -277,21 +301,35 @@ func IsTCPMetadataExchangeEnabled(node *model.Proxy) bool {
 }
 
 // ConvertLocality converts '/' separated locality string to Locality struct.
-func ConvertLocality(locality string) *core.Locality {
+func ConvertLocality(locality string) *corev3.Locality {
```

Comment:
This does not seem correct. Should n't ConvertLocality return `*core.Locality` ?

Reply:
We only use locality in CDS/EDS so I don't think we need it to return corev2. If we did, it wouldn't even compile

Reply:
Ok. Then should we just remove `ConvertLocalityV3` below?

Reply:
Thanks, yes i switched this and forgot to update it fully